ArduPlane/quadplane.cpp

```diff
@@ -4066,6 +4096,13 @@ void QuadPlane::mode_enter(void)
 {
     if (available()) {
         pos_control->set_lean_angle_max_cd(0);
+        // set rate shaping time constants
+        attitude_control->set_roll_pitch_rate_tc(acro_rp_rate_tc);
+        if (plane.control_mode == &plane.mode_qacro) {
```

Comment:
This check doesn't work because this is called before `plane.control_mode` is updated in the mode change.

You could do always set the normal one and then override the acro one here: 

https://github.com/ArduPilot/ardupilot/blob/b562d5c865a7b6b23f7c56c3f03e01c7548f398a/ArduPlane/mode_qacro.cpp#L6